client/a8c-for-agencies/sections/sites/sites-dashboard/index.tsx

```diff
@@ -138,26 +139,39 @@ export default function SitesDashboard() {
 	}, [ isLoading, isError, sitesViewState.filters, filtersMap ] );*/
 
 	useEffect( () => {
+		// If the favorites filter is set, make sure to update the filter and correctly add the is_favorite param to URLs.
+		filter.showOnlyFavorites = isFavoriteFilter;
```

Comment:
At this point, `filter.showOnlyFavorites` has the value; we don't have to set it here. (see my comment above)

Reply:
I tried that on my first approach but unfortunately it doesn't work. I suspect the reason it doesn't work is because the query params are set purely by the submenu link, whereas with everything else they are set by a field with which we can save the state when changing them. So even though 'context.query' will capture the correct value, the value is not picked up via context in `SitesDashboard` without an additional step.

Setting the `showOnlyFavorites` within `SitesDashboard` matches how we're setting search / filters etc in the same component when something changes though.

Though that said, the line in `controller.js` is redundant - `showOnlyFavorites = context.params.filter === 'favorites'` - it could be `showOnlyFavorites: isFavoriteFilter,` for example, even though it doesn't set it properly at that stage. But I can add a note to include that with any filter / search additions we do.

Reply:
Adding a note here as we had a call looking at context specifically, and the primary issue was that the `SitesDashboard` was trying to capture filter, search, sort and currentPage from SitesOverViewContext, when it was al available already in `SitesDashboardContext`. So we just needed to switch the context (which then also simplified retrieving the favorites filter as it was correctly being passed through context).